scripts/automated_ingestion/eessi_task.py

```diff
@@ -0,0 +1,1201 @@
+                    self.description.get_task_file_name())
+        next_state = self._next_state(EESSITaskState.PAYLOAD_STAGED)
+        approved_state = EESSITaskState.APPROVED
+        log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                    "next_state: '%s', approved_state: '%s'", next_state, approved_state)
+
+        default_branch_name = self.git_repo.default_branch
+        default_branch = self._get_branch_from_name(default_branch_name)
+        default_sha = default_branch.commit.sha
+        feature_branch_name = self._determine_feature_branch_name()
+        feature_branch = self._get_branch_from_name(feature_branch_name)
+        if not feature_branch:
+            # feature branch does not exist
+            # TODO: could have been merged already --> check if PR corresponding to the feature branch exists
+            # ASSUME: it has not existed before --> create it
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "branch '%s' does not exist, creating it", feature_branch_name)
+
+            feature_branch = self.git_repo.create_git_ref(f"refs/heads/{feature_branch_name}", default_sha)
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "branch '%s' created: '%s'", feature_branch_name, feature_branch)
+        else:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "found existing branch for '%s': '%s'", feature_branch_name, feature_branch)
+
+        pull_request = self._find_pr_for_branch(feature_branch_name)
+        if not pull_request:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "no PR found for branch '%s'", feature_branch_name)
+
+            # TODO: add failure handling (capture result and act on it)
+            self._update_task_states(next_state, default_branch_name, approved_state, feature_branch_name)
+
+            # TODO: add failure handling (capture result and act on it)
+            self._create_pull_request(feature_branch_name, default_branch_name)
+
+            return EESSITaskState.PULL_REQUEST
+        else:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                        "found existing PR for branch '%s': '%s'", feature_branch_name, pull_request)
+            # TODO: check if PR is open or closed
+            if pull_request.state == "closed":
+                log_message(LoggingScope.TASK_OPS, "INFO",
+                            "PR '%s' is closed, creating issue", pull_request)
+                # TODO: create issue
+                return EESSITaskState.PAYLOAD_STAGED
+            else:
+                log_message(LoggingScope.TASK_OPS_DETAILS, "INFO",
+                            "PR '%s' is open, updating task states", pull_request)
+                # TODO: add failure handling (capture result and act on it)
```

Comment:
```suggestion
                # The default branch should reflect the 'current' state. I.e. there, we write next_state into the state file
                # In the feature branch, we write approved_state into the state file so that if-and-when the PR is merged
                # into the default branch, the default branch reflects the approval of the PR.
                # TODO: add failure handling (capture result and act on it)
```